src/main/java/picard/sam/DownsampleSam.java

```diff
@@ -212,7 +220,45 @@ protected int doWork() {
         }
 
         final SamReader in = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(SamInputResource.of(INPUT));
-        final SAMFileWriter out = new SAMFileWriterFactory().makeSAMOrBAMWriter(in.getFileHeader(), true, OUTPUT);
+        final SAMFileHeader header = in.getFileHeader().clone();
+
+        if (STRATEGY == Strategy.ConstantMemory || STRATEGY == Strategy.Chained) {
+            //if running using ConstantMemory or Chained strategy, need to check if we have previously run with the same random seed
+            //collect previously used seeds
+            final int USER_SEED = RANDOM_SEED;
```

Comment:
RANDOM_SEED may be null.

Reply:
also, no reason to ALL_CAPS the userSeed variable. We usually reserve that for input arguments and statics

Reply:
null RANDOM_SEED is checked for earlier, and at this point in the code it can't be null.

Reply:
thanks for pointing that out.